sqlglot/transforms.py

```diff
@@ -348,7 +348,9 @@ def _udtf_type(u: exp.Unnest, has_multi_expr: bool) -> t.Type[exp.Func]:
                 )
             )
 
-        for join in expression.args.get("joins") or []:
+        for join in list(
+            expression.args.get("joins") or []
+        ):  # support safe removal by iterating over copy list for multiple joins
```

Comment:
This can be further optimized, e.g the first solution that comes to mind is:

```
joins = expression.args.get("joins") or []

for join in list(joins):
   ...
   joins.remove(join)
```

This way we don't have to pay the dict lookup cost at every loop iteration at L370. 

I think the `list(joins)` shallow copy is not an issue here, I would keep the solution as is.

Reply:
agree.

Reply:
updated with [d1f4219](https://github.com/tobymao/sqlglot/pull/4364/commits/d1f4219a9cb60048e3551da7f9d6d57153ab89c3)

Reply:
We must also change the L370 to use `joins`

Reply:
ohh i overlook, let me quickly fix that.